shared/gateway/gateway.go

```diff
@@ -38,53 +53,133 @@ type Gateway struct {
 	maxCallRecvMsgSize      uint64
 }
 
-// Start the gateway service. This serves the HTTP JSON traffic on the specified
-// port.
+// New returns a new gateway server which translates HTTP into gRPC.
+// Accepts a context.
+func NewValidator(
+	ctx context.Context,
+	callerId CallerId,
```

Comment:
I still don't understand why you pass the `callerId` parameter here when it's always going to be `"validator-rpc"`. How about this:
```
func NewValidator(
	ctx context.Context,
	remoteAddress,
	gatewayAddress string,
	allowedOrigins []string,
) *Gateway {
	return &Gateway{
		callerId:       "validator-rpc",
		remoteAddr:     remoteAddress,
		gatewayAddr:    gatewayAddress,
		ctx:            ctx,
		mux:            http.NewServeMux(),
		allowedOrigins: allowedOrigins,
	}
}
``` 

Similarly for the beacon node.

Reply:
Agreed. redundant.